### PR TITLE
Refactor coredata load

### PR DESCRIPTION
**Pull request by NullRa**

### CODE REVIEW

File: Common/BikeAndBusViewController.swift [hunk 1]
```diff
@@ -149,6 +152,25 @@ class BikeAndBusViewController: UIViewController {
         } catch {
         }
     }
+    // MARK: permission check
+    func checkPermission() {
```
Comment (NullRa) [inline Common/BikeAndBusViewController.swift:157]: Are you lonely? 🤔

Comment (NullRa) [reply Common/BikeAndBusViewController.swift:157]: Too lonely!

File: Common/BikeAndBusViewController.swift [hunk 2]
```diff
@@ -51,8 +53,6 @@ class BikeAndBusViewController: UIViewController {
     func uploadDefaultView() {
         //Get current location
         guard let location = MapManager.shared.manager.location else {
-            assertionFailure("Location is not ready")
-            errorAlert(title: "載入位置失敗", message: "確認定位功能已啟用", actionTitle: "OK")
```
Comment (ender503) [inline Common/BikeAndBusViewController.swift]: Why don't you show the AlertDialog? Are you also lonely? 🤔

Comment (ender503) [reply Common/BikeAndBusViewController.swift]: Sending another pull request to modify the naming from `errorAlert` to `showAlertMessage`.

Comment (NullRa) [reply Common/BikeAndBusViewController.swift]: OK

**Comment by ender503** [comment]
This branch needs to be rebased, and modify the commit message like this `Add the location permission check`, then we could merge this pull request. 😃

**Review by ender503 — APPROVED** [review]

**Pull request merged**